### PR TITLE
Remove comment about internal backwards compatibility guarantees

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: api/context/src/main/java/io/opentelemetry/context/internal/shaded/WeakConcurrentMap.java [hunk 1]
```diff
@@ -39,9 +39,6 @@
  *
  * <p>This class has been copied as is from
  * https://github.com/raphw/weak-lock-free/blob/ad0e5e0c04d4a31f9485bf12b89afbc9d75473b3/src/main/java/com/blogspot/mydailyjava/weaklockfree/WeakConcurrentMap.java
- * This is used in multiple artifacts in OpenTelemetry and while it is in our internal API,
- * generally backwards compatible changes should not be made to avoid a situation where different
- * versions of OpenTelemetry artifacts become incompatible with each other.
```
Comment (bogdandrutu) [inline api/context/src/main/java/io/opentelemetry/context/internal/shaded/WeakConcurrentMap.java:44]: Per previous discussion, we don't want to offer this guarantee, we offer compatibility with internal only if using the bom

Comment (jkwatson) [reply api/context/src/main/java/io/opentelemetry/context/internal/shaded/WeakConcurrentMap.java:44]: @anuraaga was this something you were going to address in #2654 ?

Comment (bogdandrutu) [reply api/context/src/main/java/io/opentelemetry/context/internal/shaded/WeakConcurrentMap.java:44]: The reason to remove this particular one is that it is the only place where we have this kind of requirement. I think the policy should be for all packages and at the package level

Comment (jkwatson) [reply api/context/src/main/java/io/opentelemetry/context/internal/shaded/WeakConcurrentMap.java:44]: yep...just making sure there isn't colliding work going on. :)

**Review by anuraaga — APPROVED** [review]

**Pull request merged**